pyiceberg/table/__init__.py

```diff
@@ -1578,6 +1595,120 @@ def to_ray(self) -> ray.data.dataset.Dataset:
         return ray.data.from_arrow(self.to_arrow())
 
 
+class IncrementalAppendScan(DataScan):
+    to_snapshot_id: Optional[int]
+    from_snapshot_id_exclusive: Optional[int]
+
+    def __init__(
+        self,
+        table: Table,
+        row_filter: Union[str, BooleanExpression] = ALWAYS_TRUE,
+        selected_fields: Tuple[str, ...] = ("*",),
+        case_sensitive: bool = True,
+        snapshot_id: Optional[int] = None,
```

Comment:
Incremental scan does not require the `snapshot_id` parameter, but because I made it inherit from `DataScan` , I had to include it. So I have some doubts here. How should I handle it?